pkg/pkg.mk

```diff
@@ -3,6 +3,7 @@
 #
 PKG_DIR?=$(CURDIR)
 PKG_BUILDDIR?=$(PKGDIRBASE)/$(PKG_NAME)
+PKG_SOURCE_LOCAL ?= $(PKG_$(shell echo $(PKG_NAME) | tr a-z A-Z)_SOURCE_LOCAL)
```

Comment:
I'd prefer ```PKG_SOURCE_LOCAL ?= $(PKG_SOURCE_LOCAL.$(PKG_NAME))``` to be more in line with the proposal in https://github.com/RIOT-OS/RIOT/pull/10928.
It also removes the extra subshell call just to uppercase the package name

That would also allow sth like ```foreach pkg in $USEPKG) do export $(PKG_SOURCE_LOCAL.$(PKG_NAME))```.

Unfortunately shell doesn't allow dots in variable names, so when overriding from the command line, this would have to be specified as make argument, e.g.:

```
# works:
make PKG_SOURCE_LOCAL.heatshrink=~/src/heatshrink

# doesn't work:
PKG_SOURCE_LOCAL.heatshrink=~/src/heatshrink make

# beware: doesn't work either:
PKG_SOURCE_LOCAL_CN-CBOR=foo make
```

Reply:
Hm, the command line issue is pretty unfortunate, especially as I personally prefer to use exactly the case that would not be supported anymore :-(

But keeping it consistent makes sense. Could using a `_` instead of `.` make, also in #10928, make sense maybe? In terms of parsing, it wouldn't make a difference. And also for human readability, `SOURCE_LOCAL_nimble` vs `SOURCE_LOCAL.nimble` is not the worst in the world, considering the benefits of the first option. Also `USEMODULE_INCLUDES_posix` vs `USEMODULE_INCLUDES.posix` could IMHO work, right?

Reply:
> In terms of parsing, it wouldn't make a difference.

Really? Because it is easier to find a completely different symbol as delimiter than a character that can already appear in both variable name and module name.

Reply:
I think the readability is pretty much subjective. But IMHO the key is the usability here: I would in this case surely trade (subjective) readability for versatility. Wouldn't you agree?

Reply:
I don't understand, how `SOME_VAR.vars_scope` is less readable than `SOME_VAR_vars_scope`, but as you said.. it's subjective.

Reply:
> But IMHO the key is the usability here: I would in this case surely trade (subjective) readability for versatility. Wouldn't you agree?

And no, if something is objectively more versatile but only subjectively more readable I don't agree with this trade.

Reply:
clarified offline: we both agree that usefulness is more important the readability :-)

Reply:
Yepp, and regarding compatibility of #10928: this is a very simple one-line change and #10928 was not further discussed since February. So I say: let's merge it, as it is a very useful change (for package development and in cases where you don't have Internet (e.g. on a train)). However, the point @kaspar030 made about `-` still upholds, so please provide a pattern replacement for that as well.